afqinsight/nn/pt_models.py

```diff
@@ -0,0 +1,526 @@
+            nn.Flatten(),
+            nn.Dropout(0.1),
+            nn.Linear(input_shape, 500),
+            nn.ReLU(),
+            nn.Dropout(0.2),
+            nn.Linear(500, 500),
+            nn.ReLU(),
+            nn.Dropout(0.2),
+            nn.Linear(500, 500),
+            nn.ReLU(),
+            nn.Dropout(0.3),
+            nn.Linear(500, n_classes),
+        )
+
+        if output_activation == torch.softmax:
+            self.output_activation = nn.Softmax(dim=1)
+        else:
+            self.output_activation = None
+
+        if verbose:
+            print(self.model)
+
+    def forward(self, x):
+        x = self.model(x)
+        if self.output_activation:
+            x = self.output_activation(x)
+        return x
+
+
+def mlp4_pt(input_shape, n_classes):
+    mlp4_Model = MLP4(
+        input_shape, n_classes, output_activation=torch.softmax, verbose=False
+    )
+    return mlp4_Model
+
+
+class CNN_LENET(nn.Module):
+    def __init__(
+        self, input_shape, n_classes, output_activation=torch.softmax, verbose=False
+    ):
+        super(CNN_LENET, self).__init__()
+        self.n_conv_layers = int(round(math.log(input_shape[0], 2)) - 3)
+        if verbose:
+            print(f"Pooling layers: {self.n_conv_layers}")
+
+        conv_layers = []
+        seq_len = input_shape[0]
+        in_channels = input_shape[1]
+
+        for i in range(self.n_conv_layers):
+            if i == 0:
+                conv = nn.Conv1d(
+                    in_channels=in_channels,
+                    out_channels=6,
+                    kernel_size=3,
+                    padding=1,
+                )
+            else:
+                conv = nn.Conv1d(
+                    in_channels=6 + 10 * (i - 1),
+                    out_channels=6 + 10 * i,
+                    kernel_size=3,
+                    padding=1,
+                )
+            conv_layers.append(conv)
+            conv_layers.append(nn.ReLU())
+            conv_layers.append(nn.MaxPool1d(kernel_size=2))
+            seq_len = math.floor((seq_len + 2 * 1 - 1 * (3 - 1) - 1) / 1 + 1)
+            seq_len = math.floor(seq_len / 2)
+
+        final_channels = 6 + 10 * (self.n_conv_layers - 1)
+
+        in_features = final_channels * seq_len
+
+        self.model = nn.Sequential(
+            *conv_layers,
+            nn.Flatten(),
+            nn.Linear(in_features, 120),
+            nn.ReLU(),
+            nn.Dropout(0.5),
+            nn.Linear(120, 84),
+            nn.ReLU(),
+            nn.Dropout(0.5),
+            nn.Linear(84, n_classes),
+        )
+
+        if output_activation == torch.softmax:
+            self.output_activation = nn.Softmax(dim=1)
+        else:
+            self.output_activation = None
+
+    def forward(self, x):
+        x = self.model(x)
+        if self.output_activation:
+            x = self.output_activation(x)
+        return x
+
+
+def cnn_lenet_pt(input_shape, n_classes):
+    cnn_lenet_Model = CNN_LENET(
```

Comment:
```suggestion
    """ 
    Lenet convolutional network. 
    
    Parameters 
    ----------
    input_shape : tuple 
        The shape of the input. 
    n_classes: int
        The number of classes (for classification tasks) or 
        1 (for regression tasks).
        
    Returns 
    -------
    CNN_LENET class instance.
    """
    cnn_lenet_Model = CNN_LENET(
```

Reply:
This follows the numpy docstring standard: https://numpydoc.readthedocs.io/en/latest/format.html